main/options/ConfigParser.cc

```diff
@@ -16,8 +33,8 @@ void ConfigParser::readArgsFromFile(std::shared_ptr<spdlog::logger> logger, stri
             argsPView = argsPView.substr(0, argsPView.size() - 1);
         }
         for (string_view arg : absl::StrSplit(argsPView, '\n')) {
-            if (arg.size() == 0) {
-                // skip empty line
+            if (arg.size() == 0 || isComment(arg)) {
```

Comment:
Can't you just strip the line from the position of the `#` to the end of the line?

That way you could support trailing comments too.

Reply:
No, because then if you tried to do

```
-e
"puts 'hello' # comment"
```

In the config file, it wouldn't work.

I'm 👍 to the approach trevor chose (I would have made it even more restrictive and required that the `#` be anchored at column `0`, because nowhere else in the file do we have anything that strips leading whitespace, but I don't care about that as strongly).

Reply:
I'll put an additional comment in on `isComment` explaining why we don't allow trailing comments.

Reply:
Good catch! 👍